add private network to network enum

error2215 opened the pull request.

Commit:
refactoring

Commit:
add private network to network enum

Kuly14 commented on corebc-core/src/types/network.rs:
```diff
@@ -139,7 +140,45 @@ impl Serialize for Network {
+}
+
+impl From<String> for Network {
+    fn from(s: String) -> Network {
+        match s.as_str() {
+            "mainnet" => Network::Mainnet,
+            "devin" => Network::Devin,
+            unknown => {
+                let mut parts = unknown.split('-');
+                if let Some("private") = parts.next() {
+                    if let Some(id) = parts.next() {
+                        if let Ok(id) = id.parse::<u64>() {
+                            return Network::Private(id)
+                        }
+                    }
+                }
+                panic!("Unknown network: {}", unknown);
+            }
+        }
+    }
+}
```
```suggestion
 fn from(s: String) -> Network {
    match s.as_str() {
        "mainnet" => Network::Mainnet,
        "devin" => Network::Devin,
        unknown => {
            if let ["private", id_str] = unknown.split('-').collect::<Vec<_>>().as_slice() {
                if let Ok(id) = id_str.parse::<u64>() {
                    return Network::Private(id);
                }
            }
            panic!("Unknown network: {}", unknown);
        }
    }
}
```

This makes it a bit more readable, so we don't have 3 nested if lets.

Kuly14 commented on corebc-core/src/types/network.rs:
```diff
@@ -1,36 +1,18 @@
-//      More info: <https://serde.rs/variant-attrs.html>
 //     - Add a test at the bottom of the file
 
-// We don't derive Serialize because it is manually implemented using AsRef<str> and it would
-// break a lot of things since Serialize is `kebab-case` vs Deserialize `snake_case`.
-// This means that the Network type is not "round-trippable", because the Serialize and Deserialize
-// implementations do not use the same case style.
-
```
Did you find out why is this the case? It doesn't really make sense to me to make the network type not "round-trippable"

error2215 replied:
not sure about that, but thats what I found about this 
https://github.com/gakonst/ethers-rs/issues/2038#issuecomment-1377773079
https://github.com/gakonst/ethers-rs/pull/2268#discussion_r1138931364

https://github.com/gakonst/ethers-rs/pull/2270/files
Seems like it was hot-fixed and tests for "round-trippable" were added

Kuly14 reviewed (commented):
Other than the 1 suggestion I think it looks great.

Commit:
Merge branch 'feature/private_network' of https://github.com/core-coin/corebc-rs into feature/private_network

Commit:
added test for network

Commit:
remove utils::NetworkType and replace it with types::Network

Commit:
Merge branch 'feature/private_network' of https://github.com/core-coin/corebc-rs into feature/private_network

error2215 commented:
Also I removed utils::NetworkType and replaced it with types::Network. The same logic was written twice.
Tests pass
`cargo +nightly fmt --all` and `cargo +nightly clippy` applied

Commit:
fix tests
